M2/Macaulay2/m2/html.m2

```diff
@@ -21,7 +21,7 @@ defaultHEAD = title -> HEAD splice { TITLE title, defaultCharset(), defaultStyle
 -----------------------------------------------------------------------------
 
 -- TODO: urlEncode
-htmlLiteral = s -> if s === null or not match("<|&|]]>|\42", s) then s else (
+htmlLiteral = s -> if s === null or regex("<|&|]]>|\42", s) === null then s else (
```

Comment:
Was this change needed for some reason??

Reply:
#1504.

Reply:
Is this change related to `lastMatch`?  It doesn't appear in the code.

Reply:
`lastMatch` is modified by `match`. `htmlLiteral` is called in `WebApp` mode all the time (to convert text to HTML). That means `lastMatch` can't be used in `WebApp` mode. If `lastMatch` becomes deprecated that won't be an issue, but for now I followed Mahrud's suggestion to replace `match` with `regex` (which does the exact same except doesn't update `lastMatch`).